src/syrupy/matchers.py

```diff
@@ -30,29 +34,30 @@ def path_type(
     *,
     types: Tuple["PropertyValueType", ...] = (),
     strict: bool = True,
-    regex: bool = False,
+    replacer: "Replacer" = lambda data, _: Repr(AmberDataSerializer.object_type(data)),
+    **_: Any,
 ) -> "PropertyMatcher":
     """
-    Factory to create a matcher using path and type mapping
+    Factory to create a matcher using path and type mapping.
+    Usecase:
+    Replacing all values of certain types at specified paths.
+    with their class name instead. Allows for deterministic
+    snapshots on non-deterministic class e.g. datetime, random etc.
     """
     if not mapping and not types:
         raise PathTypeError(gettext("Both mapping and types argument cannot be empty"))
 
-    def _path_match(path: str, pattern: str) -> bool:
-        if regex:
-            return re.fullmatch(pattern, path) is not None
-        return path == pattern
-
     def path_type_matcher(
         *, data: "SerializableData", path: "PropertyPath"
     ) -> Optional["SerializableData"]:
         path_str = ".".join(str(p) for p, _ in path)
         if mapping:
             for pattern in mapping:
-                if _path_match(path_str, pattern):
+                matches = re.fullmatch(pattern, path_str)
```

Comment:
The old version that checked the "regex" flag supported literal string matches. Consider trying to match the object:

```python
dictionary = {
    "this.is.a.valid.key": True,
}
```

In the new version, you'd have to escape the `.` otherwise it'll get interpreted as regex and match any character.

Reply:
True, also noticed there was no way to combine both literal string matches and regex matches in a single matcher even in the old version without escaping the literal string versions.